src/layers/basic.jl

```diff
@@ -301,7 +301,7 @@ end
     Bilinear((in1, in2) => out, σ=identity; bias=true, init=glorot_uniform)
     Bilinear(W::AbstractArray, [bias, σ])
 
-Creates a bilinear layer, which operates on two inputs at the same time.
+Creates a fully connected layer which operates on two inputs.
```

Comment:
I thought the old version was clearer

Reply:
What I would like to do is to remove all docstrings which explain "X" by saying "This makes an X", because that seems circular.

Maybe this can be better without being self-referential?

Reply:
Ah, I see that makes sense. Maybe leave in the "at the same time." part?

Reply:
Maybe like so, or perhaps too wordy?
```suggestion
Creates a fully connected layer which operates on two inputs at the same time,
with a learned weight for every triple of two input elements and an output element `x[i], y[j], z[k]`.

```

Reply:
Yeah might be too confusing. What about
```suggestion
Creates a fully connected layer which operates on two inputs at the same time,
with a separate learned weight for each input-output pair.
```

Reply:
Maybe just:
```suggestion
Creates a layer which is fully connected between two inputs and the output, and otherwise like [`Dense`](@ref).
```
"fully connected" as an adjective, since as a noun "fully connected layer" it sort-of means Dense. And leave further description to the formula two lines down.

Reply:
I think the PyTorch description is pretty short and sweet: https://pytorch.org/docs/stable/generated/torch.nn.Bilinear.html. The current docstring takes 3-4 sentences to say basically the same thing.

Reply:
That hides a lot though, as the formula is for a scalar output. But they have a stronger tradition of mysterious auto-broadcasting over extra dimensions (you are to infer that the extra dimension of A goes to the output, while those of x & y are zipped together) which their "shape" section unpacks a little. 

And I'm not sure whether saying "bilinear transformation" breaks the circular definition rule. Theirs doesn't include activation, but it does include bias, and thus isn't really linear in its inputs.

Reply:
Anyway maybe this is good enough? Sorry about the tangent, didn't want to re-write the whole thing, just regard circularity as a bug to be fixed when noticed.

Reply:
Last suggestion is good for me

Reply:
Done! Thanks all.

Reply:
I think the shape section is great and we ought to add it for all Flux layers instead of the ad-hoc current state of things, but that's a discussion for another issue. The main point is that we tend to spend a lot of prose on descriptions that could be captured with less, more structured doc text.

Reply:
Yes that's probably true. And bad prose is worse than mediocre structured info.

One simple way to do this is to add `size` examples. If you pick distinct primes for the sizes then they can fix the result, without adding another layer of "what does this notation mean?" to things, and without the "maybe that's a typo?" worry as it's tested.

In the Pytorch one, what's the distinction between ``H_in1`` and `in1_feature` exactly, are those two things or the same thing? And they resort to prose for "All but the last dimension of the inputs should be the same." which I think is saying that the * and the N mean the same thing, there is no broadcasting. (But can the N be omitted entirely? Maybe not.)